presto-docs/src/main/sphinx/optimizer/logical-planner.rst

```diff
@@ -0,0 +1,35 @@
+*****************
+Logical Planner
+*****************
+
+Presto currently has a rule based optimizer. A given SQL query goes through syntactic
+analysis, semantic analysis, logical plannning, physical planning and then execution.
+Syntax analysis or parsing is used to determine various structures in the given
+query and if they conform to the given rules. Next step is semantic analysis which
```

Comment:
The next step is semantic analysis, which gathers the semantic information from the query that is necessary for planning.  It deals with things like type checking and verifying that the column names used in the query exist for a given table.